### PR TITLE
Prepared new release: v0.17.0

### PR DESCRIPTION
**Pull request by behackl**
Release preparation PR.

Link to changelog: https://manimce--3023.org.readthedocs.build/en/3023/changelog/0.17.0-changelog.html

### CODE REVIEW

**Comment by naveen521kk** [comment]
I would like to add support for Python 3.11 (adding it to CI tests) before releasing a new version, will make a PR soon.

**Comment by naveen521kk** [comment]
As well, we should drop support for Python 3.7, since poetry is complaining
```
PS C:\dev\manim> poetry lock     
Updating dependencies
Resolving dependencies...

The current project's Python requirement (>=3.7,<3.12) is not compatible with some of the required packages Python requirement:
  - scipy requires Python >=3.8, so it will not be satisfied for Python >=3.7,<3.8
  - scipy requires Python >=3.7,<3.11, so it will not be satisfied for Python >=3.11,<3.12
  - scipy requires Python >=3.8,<3.11, so it will not be satisfied for Python >=3.7,<3.8 || >=3.11,<3.12
  - scipy requires Python >=3.8,<3.11, so it will not be satisfied for Python >=3.7,<3.8 || >=3.11,<3.12
  - scipy requires Python >=3.8,<3.12, so it will not be satisfied for Python >=3.7,<3.8
  - scipy requires Python >=3.8,<3.12, so it will not be satisfied for Python >=3.7,<3.8
  - scipy requires Python >=3.8, so it will not be satisfied for Python >=3.7,<3.8

Because no versions of scipy match >1.7.3,<1.8.0 || >1.8.0,<1.8.1 || >1.8.1,<1.9.0 || >1.9.0,<1.9.1 || >1.9.1,<1.9.2 || >1.9.2,<1.9.3 || >1.9.3,<2.0.0
 and scipy (1.7.3) requires Python >=3.7,<3.11, scipy is forbidden.
And because scipy (1.8.0) requires Python >=3.8,<3.11, scipy is forbidden.
And because scipy (1.8.1) requires Python >=3.8,<3.11
 and scipy (1.9.0) requires Python >=3.8,<3.12, scipy is forbidden.
And because scipy (1.9.1) requires Python >=3.8,<3.12
 and scipy (1.9.2) requires Python >=3.8, scipy is forbidden.
So, because scipy (1.9.3) requires Python >=3.8
 and manim depends on scipy (^1.7.3), version solving failed.

  • Check your dependencies Python requirement: The Python requirement can be specified via the `python` or `markers` properties

    For scipy, a possible solution would be to set the `python` property to ">=3.8,<3.12"
    For scipy, a possible solution would be to set the `python` property to ">=3.7,<3.11"
    For scipy, a possible solution would be to set the `python` property to ">=3.8,<3.11"
    For scipy, a possible solution would be to set the `python` property to ">=3.8,<3.11"
    For scipy, a possible solution would be to set the `python` property to ">=3.8,<3.12"
    For scipy, a possible solution would be to set the `python` property to ">=3.8,<3.12"
    For scipy, a possible solution would be to set the `python` property to ">=3.8,<3.12"

    https://python-poetry.org/docs/dependency-specification/#python-restricted-dependencies,
    https://python-poetry.org/docs/dependency-specification/#using-environment-markers
```

**Comment by behackl** [comment]
> As well, we should drop support for Python 3.7, since poetry is complaining

This is somewhat unfortunate given that Python 3.7 is still the default for Debian Buster (which is still LTS until 2024: https://wiki.debian.org/LTS), but I guess we don't really have a choice. Let's do that in #3030?

**Comment by naveen521kk** [comment]
> This is somewhat unfortunate given that Python 3.7 is still the default for Debian Buster (which is still LTS until 2024: https://wiki.debian.org/LTS)

Is that really an issue? One can always compile recent versions of Python from the source. Also, we technically have to drop it because there's no release of `scipy` which supports both `3.7` as well `3.11`.

**Comment by behackl** [comment]
> Is that really an issue?

No, not really. The Debian thing is more of a personal thought; the actual pity is that Python 3.7 is also not EOL until next year (https://endoflife.date/python) -- but well. I suppose nothing stops people from just installing an older version of Manim in that case. I'm fine with dropping 3.7. :+1:

File: docs/source/changelog/0.17.0-changelog.rst [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+Unclassified changes
+--------------------
+
+* :pr:`3035`: Include latex install instructions on ubuntu
+
+
+* :pr:`3044`: [docs] Add Debian dependencies required for pycairo and manimpango
+
+
+* :pr:`3045`: Update python-opengl -> python3-opengl for Ubuntu CI task
```
Comment (kjlubick) [inline docs/source/changelog/0.17.0-changelog.rst:265]: I'd classify this as "Changes to our development infrastructure"

File: docs/source/changelog/0.17.0-changelog.rst [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+New releases
+------------
+
+* :pr:`3023`: Prepared new release: v0.17.0
+
+
+Unclassified changes
+--------------------
+
+* :pr:`3035`: Include latex install instructions on ubuntu
```
Comment (kjlubick) [inline docs/source/changelog/0.17.0-changelog.rst:259]: Documentation change

File: docs/source/changelog/0.17.0-changelog.rst [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+
+
+* :pr:`3044`: [docs] Add Debian dependencies required for pycairo and manimpango
```
Comment (kjlubick) [inline docs/source/changelog/0.17.0-changelog.rst:262]: Documentation change

File: docs/source/changelog/0.17.0-changelog.rst (hunk 1 above)
Comment (behackl) [reply docs/source/changelog/0.17.0-changelog.rst:265]: Done, labeled https://github.com/ManimCommunity/manim/pull/3045 as infrastructure (and #3035, #3044 as documentation). Will be reflected in the next changelog generation pass.

**Comment by behackl** [comment]
I've added some descriptions to add some context to highlights and breaking changes.

Assuming the pipeline doesn't reveal any further major problems (and if there are no objections), we'll have a release later today. :partying_face:

**Review by naveen521kk — APPROVED** [review]
LGTM. I've slightly edited a few titles, like that of https://github.com/ManimCommunity/manim/pull/2925 and https://github.com/ManimCommunity/manim/pull/3044, please rerender. Thanks!

**Pull request merged**